x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/Configuration.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.session;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.query.QueryBuilder;
+
+import java.time.ZoneId;
+
+public class Configuration extends org.elasticsearch.xpack.ql.session.Configuration {
+    
+    private final String[] indices;
+    private final TimeValue requestTimeout;
+    private final String clientId;
```

Comment:
Is the `clientId` acting as a sticky session kind of identifier?

Reply:
Not really sticky however I imagined it would be useful for metrics such as knowing whether we've been called by SIEM or some different entity.